cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -284,6 +284,49 @@ const char* reverse_utf8(gdv_int64 context, const char* data, gdv_int32 data_len
   return ret;
 }
 
+// Trim a utf8 sequence
+FORCE_INLINE
+const char* trim_utf8(gdv_int64 context, const char* data, gdv_int32 data_len,
+                      int32_t* out_len) {
+  if (data_len == 0) {
+    *out_len = 0;
+    return "";
+  }
+
+  gdv_int32 start = 0, end = data_len - 1;
+  // start and end denote the first and last positions of non-space
+  // characters in the input string respectively
+  while (start <= end && data[start] == ' ') {
+    ++start;
+  }
+  while (end >= start && data[end] == ' ') {
```

Comment:
for utf8 multi-byte characters, is it possible that the last byte looks like a space but it's actually part of a multi-byte character ?

Reply:
np
0x20 is always single-byte space. c.f. https://en.wikipedia.org/wiki/UTF-8

Reply:
Verified that for utf8, 0x20 is always a single-byte space and is not part of any multi-byte characters.